pkg/monitortests/node/legacynodemonitortests/kubelet.go

```diff
@@ -238,64 +246,105 @@ func testContainerFailures(adminRestConfig *rest.Config, events monitorapi.Inter
+				} else if len(messages) >= maxRestartCountForFlakes {
+					excessiveExitsByNamespaceForFlakeTests[namespace] = append(excessiveExitsByNamespaceForFlakeTests[namespace], fmt.Sprintf("%s restarted %d times at:\n%s", locator, len(messages), strings.Join(messageSet.List(), "\n")))
+				}
 			}
 		}
 	}
-	sort.Strings(excessiveExits)
+	for namespace, excessiveExitsFails := range excessiveExitsByNamespaceForFailedTests {
+		sort.Strings(excessiveExitsFails)
+		excessiveExitsByNamespaceForFailedTests[namespace] = excessiveExitsFails
+	}
+	for namespace, excessiveExitsFlakes := range excessiveExitsByNamespaceForFlakeTests {
+		sort.Strings(excessiveExitsFlakes)
+		excessiveExitsByNamespaceForFlakeTests[namespace] = excessiveExitsFlakes
+	}
 
 	var testCases []*junitapi.JUnitTestCase
 
-	const failToStartTestName = "[sig-architecture] platform pods should not fail to start"
-	if len(failures) > 0 {
-		testCases = append(testCases, &junitapi.JUnitTestCase{
-			Name:      failToStartTestName,
-			SystemOut: strings.Join(failures, "\n"),
-			FailureOutput: &junitapi.FailureOutput{
-				Output: fmt.Sprintf("%d container starts had issues\n\n%s", len(failures), strings.Join(failures, "\n")),
-			},
-		})
+	for _, namespace := range sets.List(openshiftNamespaces) { // this ensures we create test case for every namespace, even in success cases
+		failures := failuresByNamespace[namespace]
+		failToStartTestName := fmt.Sprintf("[sig-architecture] platform pods in ns/%s should not fail to start", namespace)
+		if len(failures) > 0 {
+			testCases = append(testCases, &junitapi.JUnitTestCase{
+				Name:      failToStartTestName,
+				SystemOut: strings.Join(failures, "\n"),
+				FailureOutput: &junitapi.FailureOutput{
+					Output: fmt.Sprintf("%d container starts had issues\n\n%s", len(failures), strings.Join(failures, "\n")),
+				},
+			})
+		}
+		// mark flaky for now while we debug
+		testCases = append(testCases, &junitapi.JUnitTestCase{Name: failToStartTestName})
 	}
-	// mark flaky for now while we debug
-	testCases = append(testCases, &junitapi.JUnitTestCase{Name: failToStartTestName})
 
 	// We want to deflake this test.
 	// Plan is to release this test and report any failures for pods
 	// that restart more than 3 times
 	// We will then build exclusion rules for those that we see
 	// and then make this test fail for any case that doesn't match the rules
 	// we have.
-	const excessiveRestartTestName = "[sig-architecture] platform pods should not exit more than once with a non-zero exit code"
-	if len(excessiveExits) > 0 {
-		testCases = append(testCases, &junitapi.JUnitTestCase{
-			Name:      excessiveRestartTestName,
-			SystemOut: strings.Join(excessiveExits, "\n"),
-			FailureOutput: &junitapi.FailureOutput{
-				Output: fmt.Sprintf("%d containers with multiple restarts\n\n%s", len(excessiveExits), strings.Join(excessiveExits, "\n\n")),
-			},
-		})
```

Comment:
need to keep our enforcement.  We don't want to lose this while people are getting comfortable.

Reply:
done. Added the success case also to not break test frameworks.